birdhouse/docker-compose.yml

```diff
@@ -1,4 +1,12 @@
-version: "2.1"
+version: "3.4"
+
+x-logging:
+  &default-logging
+  driver: "json-file"
+  options:
+    max-size: "200k"
```

Comment:
So we have not set any of those config currently (driver, max-size, max-file).  What's was their default?

It's very nice to have a unified logging config for all containers but can each container, if needed, not use this config and roll their own?

And to loop back to the original Weaver PR, Weaver need to "hook" into other components logs and that's why it prefers other components to have unified logging config or is there another reason to introduce this unified logging?

Just to be clear, I like it to have a default and unified logging config, I just want to understand its relation to Weaver.

Reply:
Docker's default is json so that remains the same.
The difference here is that we add log rotation over 10 files. 
This is a config we have found a long time ago when deploying on our instances since twitcher and magpie can generate an immense quantity of logs (on each request). This allows discarding them gradually instead of accumulating GB of useless logs. 
Each container can define its own logging section by using the relevant config instead of the `default-logging` reference. 
I don't think there is a nice way to automate that though. Best could be to have an optional-component, but I think it is better to control the logs.

Reply:
Using Weaver, it is a bit more tricky since a different docker-compose file is employed. 
Depending on how docker-compose merges the files, the reference could or won't work across files. 
Basically, if files are merged, and then references resolved, it would work. But if internally it uses PyYAML or similar lib, to read each file individually and merge them, Weaver's compose will raise a missing reference. So the config would need to be duplicated in its own file.

Reply:
So if I understand correctly, Weaver did not need this central logging and the log rotation to avoid accumulating infinite logs (which is indeed a nice thing).

From our recent "contributing guideline", this PR should not have been mixed with the Weaver PR to begin with.  Or if mixed in, it should have been "well explained" that it is not required by Weaver?

Reply:
Yes, this is why it has been split by itself now that we updated the contribution guideline. 
Please consider that Weaver PR was started in January and review only started in March, both long before we even started discussions of contribution guidelines.